src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/Native/SafeHandles.cs

```diff
@@ -12,22 +13,19 @@ namespace Internal.Cryptography.Pal.Native
     /// </summary>
     internal abstract class SafePointerHandle<T> : SafeHandle where T : SafeHandle, new()
     {
-        public SafePointerHandle()
+        protected SafePointerHandle()
             : base(IntPtr.Zero, true)
         {
         }
 
-        public sealed override bool IsInvalid
+        internal static T CreateInvalid()
```

Comment:
:thought_balloon: There are a few problems with treating the method as a factory and thus placing it here. The primary problem is there is no way to define "factory" method such that it can be identified by an analyzer. Special cases like this make it difficult to impossible to transition to using analyzers for improving the overall consistency within source files. For example, it is very easy to check the order Constructors &rarr; Properties &rarr; Methods, but as soon as you allow any non-constructor "factory" method above a property, the best we can do is Constructors &rarr; `static` Methods &rarr; Properties &rarr; More `static` Methods &rarr; Methods. This is a more complicated structure for developers to recognize and adapt to, and requires more complicated analyzers and documentation. I believe special treatment for "factory" methods within the layout of a file raises more questions than it answers.